### ISSUE TITLE
Uncaught thrown terms cause Espec runner to crash

### ISSUE THREAD

**Issue by jonathanperret** [antonmi/espec#80]
Given the following spec:

```elixir
defmodule ThrowIssue do
  use ESpec

  it do
    throw :some_term
  end
end
```

I would expect Espec to report one failing example. Instead, here is what happens:

```

=ERROR REPORT==== 17-Feb-2016::17:41:46 ===
** Generic server 'Elixir.ESpec.Runner' terminating
** Last message in was run
** When Server state == #{opts => [{finally,#Fun<elixir_compiler_1.2.64185228>},
                           {finish_loading_time,{1455,727306,893861}},
                           {file_opts,[{<<"/Users/jonathanperret/src/espec/throw_issue.exs">>,
                                        []}]},
                           {start_loading_time,{1455,727306,820606}},
                           {before,#Fun<elixir_compiler_1.1.64185228>},
                           {included_applications,[]}],
                          specs => ['Elixir.ThrowIssue']}
** Reason for termination ==
** {bad_return_value,some_term}
** (EXIT from #PID<0.47.0>) bad return value: :some_term
```

**Comment by antonmi** [comment]
Thanks!
I guess expected behaviour is to fail with message:
"throw :some_term"

**Comment by antonmi** [comment]
The fix will be available in the next ESpec version

**Comment by antonmi** [comment]
The fix is included into 0.8.12

**Comment by jonathanperret** [comment]
Looks great, thanks!
